### ISSUE TITLE
Russian Dictionaries Highlight All Words in Red Despite Being Recognized in Suggestions

### ISSUE THREAD

**Issue by NagBoard** [futo-org/android-keyboard#92]
**Phone:** Redmi Note 12 Pro
**Android version:** 11 RKQ1.200826.002
**MIUI version:** MIUI Global 13.0.3(RHGEUXM) Stable
**Keyboard:** Russian, East Slavic
**FUTO Keyboard version:** Version 0.1.22-playstore
**Steps to reproduce:** Just type any word that is recognized by the dictionary in suggestions bar, it will still highlight red even though the dictionary recognizes the word in the suggestions bar.

Originally I thought that issue was caused by 3rd party dictionaries, so far I've tried several different and they all had the same issue, and even when I switch to a built-in dictionary and use words that are recognized by the suggestion bar when using built-in dictionary the issue still persists. In attached photo I used 3rd party dictionary attached below and just pressed middle suggestion multiple types, not even writing anything myself — all words were pulled up from the dictionary. GBoard is turned off as an input method.

**Photo:** 
![Russian dictionaries highlight everything in red](https://github.com/futo-org/android-keyboard/assets/100212155/e0b3b2e3-ff89-46c3-a3ea-0565244b77ba)

**3rd party dictionary:** https://codeberg.org/Helium314/aosp-dictionaries/src/branch/main/dictionaries/main_ru.dict

**Comment by NagBoard** [comment]
Upon further investigation, it just seems like the UI bug only. It seems like the dictionaries that I've imported work pretty well and they consistently give me meaningful suggestions. Furthermore, when I am adding new words to the build-in dictionary that came with the program itself, it does add those new words to the dictionary pool, but they are still highlighted red. My guess is that any word that was not shipped, built in to the keyboard itself, when added as a new word, will appear at the suggestions bar and functionally will behave as expected, but UI will highlight it in red regardless.

**Comment by ragingdemon** [comment]
This also happens with Spanish, it doesn't highlight all red like in your example but a lot of it. My guess is that only the English dictionary is being use to determine whether words are correct or not.

**Comment by NagBoard** [comment]
> This also happens with Spanish, it doesn't highlight all red like in your example but a lot of it. My guess is that only the English dictionary is being use to determine whether words are correct or not.

Yep, pretty much sums up what I've posted above but more eloquently.

**Comment by andrej-dyck** [comment]
Same with German

**Comment by NagBoard** [comment]
Issue persists on newest version.

**Comment by moustaleurie** [comment]
Same issue with greek too. Also, while I can ignore the red lines for now, when I want to change a word using the suggestions futo provides, the pop up dialogue to "add to dic" or "delete" gets in the way!

**Comment by dean-futo** [comment]
#145 is a duplicate - merged that issue into this one.

**Comment by dean-futo** [comment]
Others have mentioned that even through 3rd party dictionaries are working as intended, the keyboard may be referring to its built-in dictionary when determining if something is spelled correctly or not. @abb128 can you confirm what might be going on?

**Comment by Guyc1800** [comment]
I have the same issue, and from my testing it seems when the system wide language is different then the language you're typing all the words has red underline. at first Hebrew had this issue when system wide language was English(and typing in English worked fine), then I swapped the system language to Hebrew and typing in Hebrew worked fine but English didn't.
Also for some reason on certain apps and site it highlights red and others not, from my experience on WhatsApp(version 2.24.24.79) and on youtube search bar with chrome(version131.0.6778.135) it has this problem consistently.
Another thing is for some reason some words that used to have this problem suddenly doesn't get highlighted(have no idea why) so to reproduce it is suggested to use a set of words and change the set from time to time.
One last thing is currently a workaround I found is to turn off the "Auto-correction" in "Keyboard & Typing" on the app settings. 
I hope this will helps
futo keyboard v. 0.1.24.1-rc1-playstore

**Comment by NagBoard** [comment]
> I have the same issue, and from my testing it seems when the system wide language is different then the language you're typing all the words has red underline. at first Hebrew had this issue when system wide language was English(and typing in English worked fine), then I swapped the system language to Hebrew and typing in Hebrew worked fine but English didn't.
 

> **The system language**

May be this??

**Comment by NagBoard** [comment]
@dean-futo 


![Screenshot_2025-02-06-23-54-59-806_com.android.settings.jpg](https://github.com/user-attachments/assets/fe4e7b5b-9bc9-4eab-a0e0-bbc800e13717)

Was skimming through options and went to dictionary. I only have one! I use both Russian and English keyboards, but the Russian dictionary is missing. On the figure above you can see that custom-dictionary words are clumped in the English  dictionary. 

I also have some `xx (YY)` dictionary (have no idea what it is) and `For all languages` dictionary. Both are empty.

**Comment by themngn** [comment]
The problem is that the keyboard doesn't communicate language to the system (languageTag is responsible for that, from what I could find), so it underlines words as incorrect because the system thinks that the user types in English regardless of the layout selected on the keyboard. 

This issue should be categorized into "High" or preferably "Very High Priority" because it's an extremely annoying bug, that affects most/all bilingual users.

@abb128

**Comment by Radoslaw7** [comment]
Same with Polish

**Comment by wertzuz** [comment]
Still existing in German

**Comment by NagBoard** [comment]
@dean-futo, any updates on this one?

**Comment by Hanqnero** [comment]
still the same in russian. Is there a way to disable spellcheck for now?

**Comment by abb128** [comment]
I'm going to close this in favor of #1334 which tracks the exact solution and not the problem
